### DISCUSSION TITLE
Bulk edit queries

### DISCUSSION THREAD

**Discussion by Auric-Manteo** [getredash/redash#5928] [Ideas]
I would like to add a new data source with the exact same data and change all queries that use the old data source to the new one.
Is there some bulk edit functionality?
For now I can make these changes directly in the database, but would prefer an option in the UI.
This would be super helpful to search and replace similar snippets as well.

**Comment by arikfr** [comment]
Adding bulk edit functionality might be a big lift, but adding the option within the context of a data source to "Update all queries using this data source to use a different one" should be much simpler task.

**Comment by arikfr** [comment]
Btw, besides doing the change directly in the database, you can also use the API to achieve the same.

**Reply by Auric-Manteo** [reply]
That sounds like an excellent alternative. Thanks for the pointer!

**Comment by RobinDarrah** [comment]
While this isn't related to your specific query, if other people come here looking for how to bulk edit queries be careful of editing them directly in the database - as without updating query hash values as well you'll get into a situation where results are shown for the previous query version.
